BlueprintUI/Sources/Element/ElementContent.swift

```diff
@@ -111,6 +111,65 @@ extension ElementContent {
 }
 
 
+extension ElementContent {
+
+    /// Creates a new `ElementContent` which uses the provided element to measure its
+    /// size, but does not place the element as a child in the final, laid out hierarchy.
+    ///
+    /// This is useful if you are placing the element in a nested `BlueprintView`, for example (eg
+    /// to create a stateful element) and just need this element to be correctly sized.
+    public init(byMeasuring element: Element) {
+        storage = MeasureNestedElementStorage(element: element)
+    }
+
+    private struct MeasureNestedElementStorage: ContentStorage {
+
+        let element: Element
+
+        let childCount: Int = 0
+
+        func measure(
+            in constraint: SizeConstraint,
+            with environment: Environment,
+            state: ElementState
+        ) -> CGSize {
+
+            let childState = state.childState(
+                for: element,
+                in: environment,
+                with: .identifier(for: element, key: nil, count: 1),
+                kind: .measurementOnly
+            )
+
+            precondition(type(of: element) == type(of: childState.element.latest))
+
+            return childState.measure(in: constraint, with: environment) { environment in
+                childState.elementContent.measure(in: constraint, with: environment, state: childState)
+            }
+        }
+
+        func performLayout(
+            in size: CGSize,
+            with environment: Environment,
+            state: ElementState
+        ) -> [LayoutResultNode] {
+            []
+        }
+
+        func forEachElement(
```

Comment:
How is this used?

Reply:
We use this to enumerate the element tree, to make sure that the newest element makes it into the live state; even if we return a cached layout, eg: https://github.com/square/Blueprint/pull/404/files#diff-87a9ff4712c4ac0a865f6587fa32cecfb76335e0767e569de804727956baffc4L489